saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/authentication/SimpleSaml2AuthenticatedPrincipal.java

```diff
@@ -27,13 +29,25 @@
 class SimpleSaml2AuthenticatedPrincipal implements Saml2AuthenticatedPrincipal, Serializable {
 
 	private final String name;
+	private final Map<String, Object> attributes;
 
 	SimpleSaml2AuthenticatedPrincipal(String name) {
```

Comment:
Is there still a use case in the application for this constructor? Since this is a package-private class, we should consider removing it.